### PR TITLE
Consistent output directories

### PR DESCRIPTION
**Pull request by jdahm**
This PR puts the tests and benchmarks in a directory structure that mimics the source.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -93,11 +93,11 @@ install(EXPORT umpire-targets DESTINATION share/umpire/cmake)
 add_subdirectory(src)
 
 if (ENABLE_TESTS)
-  set (UMPIRE_TEST_OUTPUT_DIR ${CMAKE_BINARY_DIR}/test)
+  set (UMPIRE_TEST_OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY})
```
Comment (davidbeckingsale) [inline CMakeLists.txt]: Is this a CMake variable?

Comment (jdahm) [reply CMakeLists.txt]: It seems to be something defined by BLT.
```
~/Umpire/blt $ git grep TEST_OUTPUT_DIRECTORY
SetupBLT.cmake:    set(TEST_OUTPUT_DIRECTORY
```

Comment (davidbeckingsale) [reply CMakeLists.txt]: Perfect!

**Comment by davidbeckingsale** [comment]
@jdahm - let's wait for @mcfadden8 to review this prior to merging.

**Comment by jdahm** [comment]
These are put under the root directory, but they could be put under `bin/`, so the output directories would be `bin/{examples,tools,...}`. What do you think?

**Comment by davidbeckingsale** [comment]
I think nesting the directories like that is a good idea.

File: CMakeLists.txt [hunk 2]
```diff
@@ -93,19 +93,21 @@ install(EXPORT umpire-targets DESTINATION share/umpire/cmake)
 add_subdirectory(src)
 
 if (ENABLE_TESTS)
-  set (UMPIRE_TEST_OUTPUT_DIR ${CMAKE_BINARY_DIR}/test)
+  set(UMPIRE_TESTS_OUTPUT_DIR ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests)
```
Comment (davidbeckingsale) [inline CMakeLists.txt]: I'm assuming `CMAKE_RUNTIME_OUTPUT_DIRECTORY` will be `bin`?

File: benchmarks/CMakeLists.txt [hunk 3]
```diff
@@ -12,29 +12,31 @@
 # For details, see https://github.com/LLNL/Umpire
 # Please also see the LICENSE file for MIT license.
 ##############################################################################
+set(benchmarks_dir ${UMPIRE_BENCHMARKS_OUTPUT_DIR})
```
Comment (davidbeckingsale) [inline benchmarks/CMakeLists.txt]: I don't think renaming the variable here helps much? Why not use it directly.

File: CMakeLists.txt (hunk 2 above)
Comment (jdahm) [reply CMakeLists.txt]: Yeah, it's another variable defined by BLT

File: benchmarks/CMakeLists.txt (hunk 3 above)
Comment (jdahm) [reply benchmarks/CMakeLists.txt]: 👍

**Comment by davidbeckingsale** [comment]
@mcfadden8 - can you take a look at this?

File: tests/integration/replay/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+replay_test?.csv
```
Comment (mcfadden8) [inline tests/integration/replay/.gitignore]: Missing a newline at end of file

Comment (jdahm) [reply tests/integration/replay/.gitignore]: Good catch. Thanks!

**Comment by davidbeckingsale** [comment]
Merge at will :)

**Pull request merged**